client/config.c

```diff
@@ -26,6 +29,49 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+static uint32_t TDNFParseOSInfo(PTDNF_CONF pConf)
+{
+    char temp[USERAGENTHEADERMAXLENGTH];
+    char name[10];
+    char version[10];
+    int name_found = 0;
+    uint32_t dwError = 0;
+    FILE *file = NULL;
+
+    file = fopen(OSCONFFILE, "r");
+
+    if (!file) {
+        dwError = errno;
+        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+    }
+
+    while (fgets(temp, USERAGENTHEADERMAXLENGTH, file)) {
```

Comment:
Use `getline` instead. Example here https://stackoverflow.com/questions/3501338/c-read-file-line-by-line

Reply:
`fgets()` is just fine.

Reply:
`fgets` works here but we assume that max line length is 50 bytes, with `getline` we don't need not make any assumptions and fits perfectly for this use case.

Reply:
50 characters is too short, we can just create a 256 byte buffer on the stack, and no serious `/etc/os-release` file will have lines any longer, and even if we can skip the rest. We can use `getline()`, I just think it's overkill.